File: BlogModule.Application.Contract/BlogPostComments/GetBlogPostComments/GetBlogPostCommentsResponse.cs
using System.Collections.Immutable;
using BlogModule.Domain.Enums;
using BlogModule.Domain.Models;
using Results.Contract;
using Results.Domain;

namespace BlogModule.Application.Contract.BlogPostComments.GetBlogPostComments;

public class GetBlogPostCommentsResponse: IApiCollectionResult<BlogPostCommentModel>, IPageableResponse, ISortable<BlogPostCommentSortColumnEnum?>, ISuccessResult
{
    public int? Page { get; init; }
    public int? PageSize { get; init; }
    public int TotalCount { get; init; }
    public BlogPostCommentSortColumnEnum? OrderBy { get; init; }
    public SortOrderEnum? OrderDirection { get; init; }
    public ImmutableArray<BlogPostCommentModel> Items { get; init; }
}

File: BlogApi/Application.contract/BlogPostComments/GetBlogPostComments/get_blog_post_comments_response.go
package contract

import (
	"identity-api/Domain/models"
	"identity-api/Domain/pagination"
	responses "identity-api/Domain/responses"
)

type GetBlogPostCommentsResponse struct {
	responses.BaseResponse
	pagination.PaginatedResult[models.BlogPostCommentModel] `json:",inline"`
}

type GetBlogPostCommentsResponse200 struct {
	Success        bool                       `json:"success" example:"true" swaggertype:"boolean"`
	Message        string                     `json:"message" example:"Comments retrieved successfully" swaggertype:"string"`
	Page           *int                       `json:"page" example:"1" swaggertype:"integer"`
	PageSize       *int                       `json:"pageSize" example:"10" swaggertype:"integer"`
	TotalCount     int                        `json:"totalCount" example:"100" swaggertype:"integer"`
	OrderBy        *BlogPostCommentSortColumn `json:"orderBy" example:"dateAdded" swaggertype:"string"`
	OrderDirection *SortOrder                 `json:"orderDirection" example:"desc" swaggertype:"string"`
	Items          []BlogPostCommentModel     `json:"items" swaggertype:"array,object"`
}

type GetBlogPostCommentsResponse400 struct {
	Success bool   `json:"success" example:"false" swaggertype:"boolean"`
	Message string `json:"message" example:"Invalid request parameters" swaggertype:"string"`
}

type GetBlogPostCommentsResponse404 struct {
	Success bool   `json:"success" example:"false" swaggertype:"boolean"`
	Message string `json:"message" example:"Blog post not found" swaggertype:"string"`
}

type GetBlogPostCommentsResponse500 struct {
	Success bool   `json:"success" example:"false" swaggertype:"boolean"`
	Message string `json:"message" example:"Internal server error while retrieving comments" swaggertype:"string"`
}
